src/scraper/scraper.cpp

```diff
@@ -5434,7 +5434,91 @@ UniValue archivelog(const UniValue& params, bool fHelp)
     return UniValue(ret);
 }
 
+UniValue convergencereport(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() > 0 )
+        throw std::runtime_error(
+                "convergencereport\n"
+                "Display local node report of scraper convergence.\n"
+                );
+
+    // See if converged stats/contract update needed...
+    bool bConvergenceUpdateNeeded = true;
+    {
+        LOCK(cs_ConvergedScraperStatsCache);
+
+        if (GetAdjustedTime() - ConvergedScraperStatsCache.nTime < (nScraperSleep / 1000) || ConvergedScraperStatsCache.bClean)
+        {
+            bConvergenceUpdateNeeded = false;
+        }
+    }
+
+    if (bConvergenceUpdateNeeded)
+    {
+        // Don't need the output but will use the global cache, which will be updated.
+        ScraperGetSuperblockContract(false, false);
+    }
+
+    UniValue result(UniValue::VOBJ);
+
+    {
+        LOCK(cs_ConvergedScraperStatsCache);
+
+        if (!ConvergedScraperStatsCache.NewFormatSuperblock.WellFormed())
+        {
+            UniValue error(UniValue::VOBJ);
+            error.pushKV("Error:", "A convergence cannot be formed at this time. Please review the scraper.log for details.");
```

Comment:
Maybe better to `throw JSONRPCError(RPC_MISC_ERROR, "...")` rather than returning the message in a success response?

Reply:
Yeah

Reply:
A lot of the legacy RPC's don't do this either. We need to clean this up across the board.